src/array-repeat-strategy.js

```diff
@@ -20,16 +20,82 @@ export class ArrayRepeatStrategy {
   * @param items The new array instance.
   */
   instanceChanged(repeat, items) {
-    if (repeat.viewsRequireLifecycle) {
-      let removePromise = repeat.removeAllViews(true);
-      if (removePromise instanceof Promise) {
-        removePromise.then(() => this._standardProcessInstanceChanged(repeat, items));
+    const itemNameInBindingContext = repeat.local;
+    
+    const children = repeat.viewSlot.children;
```

Comment:
Use the AbstractRepeater's [viewCount](https://github.com/aurelia/templating-resources/blob/master/src/repeat.js#L209) method here